File: android/app/src/main/java/com/rbdarts/feature/matchscoring/LeagueMatchDiagnostics.kt
package com.rbdarts.feature.matchscoring

import com.rbdarts.core.observability.DiagnosticEvent
import com.rbdarts.core.observability.DiagnosticSeverity
import com.rbdarts.core.observability.DiagnosticsReporter
import com.rbdarts.core.observability.PrivacySafeDiagnostics

class LeagueMatchDiagnostics(
    private val reporter: DiagnosticsReporter = PrivacySafeDiagnostics()
) {
    fun recordPermissionDenied(action: String) {
        reporter.record(DiagnosticEvent("league_permission_denied", DiagnosticSeverity.Warning, mapOf("action" to action)))
    }

    fun recordLockTransition(gameId: String) {
        reporter.record(DiagnosticEvent("league_game_locked", DiagnosticSeverity.Info, mapOf("gameId" to gameId)))
    }

    fun recordFinalizeFailure(matchId: String) {
        reporter.record(DiagnosticEvent("league_finalize_failed", DiagnosticSeverity.Error, mapOf("matchId" to matchId)))
    }
}

File: ios/RBDarts/Features/MatchScoring/LeagueMatchDiagnostics.swift
import Foundation

public struct LeagueMatchDiagnostics {
    private let reporter: DiagnosticsReporting

    public init(reporter: DiagnosticsReporting = PrivacySafeDiagnostics()) {
        self.reporter = reporter
    }

    public func recordPermissionDenied(action: String) {
        reporter.record(DiagnosticEvent(name: "league_permission_denied", severity: .warning, attributes: ["action": action]))
    }

    public func recordLockTransition(gameId: String) {
        reporter.record(DiagnosticEvent(name: "league_game_locked", severity: .info, attributes: ["gameId": gameId]))
    }

    public func recordFinalizeFailure(matchId: String) {
        reporter.record(DiagnosticEvent(name: "league_finalize_failed", severity: .error, attributes: ["matchId": matchId]))
    }
}
